contracts/SortitionPool.sol

```diff
@@ -73,29 +121,26 @@ contract SortitionPool is SortitionTree, Ownable {
   {
     uint256 weight = getWeight(authorizedStake);
 
+    uint32 id = getOperatorID(operator);
+    Rewards.updateOperatorRewards(id, uint32(weight));
+
     if (weight == 0) {
       _removeOperator(operator);
     } else {
       updateOperator(operator, weight);
     }
   }
 
-  /// @notice Removes an operator from the pool.
-  /// @dev Can be called only by the contract owner.
-  /// @param operator Address of the operator to be removed.
-  function removeOperator(address operator) public onlyOwner onlyUnlocked {
-    _removeOperator(operator);
-  }
-
-  /// @notice Ban rewards for given operators for given period of time.
-  /// @dev Can be called only by the contract owner.
-  /// @param operators IDs of banned operators.
-  /// @param duration Duration of the ban in seconds.
-  function banRewards(uint32[] calldata operators, uint256 duration)
-    external
+  function setRewardIneligibility(uint32[] calldata operators, uint256 until)
+    public
     onlyOwner
   {
-    // TODO: Implementation
+    Rewards.setIneligible(operators, until);
```

Comment:
Can we emit an event here and in `restoreRewardEligibility`?

Reply:
Would the following be suitable?
```
  event IneligibleForRewards(
      uint32[] ids,
      uint256 until
  );

  event EligibilityRestored(
      address operator,
      uint32 id
  );
```
What fields in the events should be indexed, if any?

Reply:
I think they make sense. I would both fields in `EligibilityRestored`. Indexing in `IneligibleForRewards` does not make sense IMO unless we emit an event inside the inner loop in `Rewards` library which could consume more gas.